### PR TITLE
fix(amqp sink): remove unnecessary unwrap & emit event dropped errors

### PR DESCRIPTION
**Pull request by dsmith3197**
Fixes: https://github.com/vectordotdev/vector/issues/18921

### CODE REVIEW

File: src/sinks/amqp/sink.rs [hunk 1]
```diff
@@ -43,6 +43,7 @@ impl AmqpSink {
             .await
             .map_err(|e| BuildError::AmqpCreateFailed { source: e })?;
 
+        // Enable confirmations on the channel.
```
Comment (dsmith3197) [inline src/sinks/amqp/sink.rs:46]: This only needs to be called once: https://www.rabbitmq.com/confirms.html#publisher-confirms

File: src/sinks/amqp/service.rs [hunk 2]
```diff
@@ -128,11 +126,8 @@ impl Service<AmqpRequest> for AmqpService {
             match fut {
                 Ok(result) => match result.await {
                     Ok(lapin::publisher_confirm::Confirmation::Nack(_)) => {
-                        warn!("Received Negative Acknowledgement from AMQP server.");
-                        Ok(AmqpResponse {
-                            json_size: req.metadata.into_events_estimated_json_encoded_byte_size(),
-                            byte_size,
-                        })
+                        emit!(AmqpNackError);
+                        Err(AmqpError::AmqpNack)
```
Comment (dsmith3197) [inline src/sinks/amqp/service.rs:130]: A negative acknowledgement should be treated as a failure.

File: src/sinks/amqp/sink.rs (hunk 1 above)
Comment (neuronull) [reply src/sinks/amqp/sink.rs:46]: good find!

File: src/sinks/amqp/service.rs (hunk 2 above)
Comment (neuronull) [reply src/sinks/amqp/service.rs]: Should we document that somewhere? Either a code comment here or in a design spec location.

**Review by neuronull — APPROVED** [review]

**Review by bruceg — APPROVED** [review]
Makes sense to me.

File: src/sinks/amqp/service.rs (hunk 2 above)
Comment (dsmith3197) [reply src/sinks/amqp/service.rs]: I would argue that emitting/returning an error here is self-documenting.  To be clear, this change is simply improving the observability of this component, giving the user insight into the number of Nacks that occur.

**Pull request merged**